src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,100 @@
+package duke;
+
+import duke.command.*;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class that parses user input and returns the corresponding command.
+ */
+public class Parser {
+
+    /**
+     * Main method that takes in raw user input and returns the corresponding command.
+     *
+     * @param fullCommand The raw string input by the user.
+     * @return The command corresponding to the user's input.
+     * @throws DukeException If user's input is invalid (does not correspond to any command or invalid parameters).
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] input = fullCommand.split(" ", 2); //return [commandWord, restOfCommand]
```

Comment:
Should the variable be plural instead?

input seems to be an array, according to the coding standards it should be plural.